influence_benchmark/utils/wandb_logging.py

```diff
@@ -143,11 +157,34 @@ def log_iteration_data_to_wandb(iteration_step, top_n_trajs_per_initial_state, t
         commit=True,
     )
     traj_timesteps_df = load_trajectories(traj_iter_dir)
-    avg_rew_df = compute_average_traj_rewards(traj_timesteps_df)
-    traj_timesteps_df = traj_timesteps_df.merge(avg_rew_df, on=["env_name", "initial_state_id", "trajectory_id"])
-    trajectories = extract_wandb_data(traj_timesteps_df)
+
+    if final_reward:
+        rew_df = compute_final_turn_rewards(traj_timesteps_df)
+    else:
+        rew_df = compute_average_traj_rewards(traj_timesteps_df)
+
+    traj_timesteps_df = traj_timesteps_df.merge(rew_df, on=["env_name", "initial_state_id", "trajectory_id"])
+    trajectories = extract_wandb_data(traj_timesteps_df, final_reward)
     # Shuffle the trajectories in the df
     random.shuffle(trajectories)
+
+    # Calculate mean reward and influence for each environment
+    env_stats = defaultdict(lambda: {"rewards": [], "influences": []})
+    for trajectory in trajectories:
+        env_name = trajectory["env_name"]
+        env_stats[env_name]["rewards"].append(trajectory["avg_reward"])
+        env_stats[env_name]["influences"].append(trajectory["avg_influence"])
```

Comment:
We should probably move this to the end of "extract_wandb_data" and have it return both `trajectories`, and `env_states`. I know this is my code, but if you could make this change together with yours, that would be helpful